projects/storefrontlib/cms-components/misc/icon/icon.component.ts

```diff
@@ -41,87 +42,97 @@ export class IconComponent {
    * The cxIcon directive is bound to the icon type. You can feed the `ICON_TYPE` to
    * accomplish a configurable button in the UI.
    */
-  @Input() set cxIcon(type: ICON_TYPE) {
+  @Input() set cxIcon(type: ICON_TYPE_STRING) {
     this.setIcon(type);
   }
 
   /**
    * The type input parameter is bound to the icon type. You can feed the `ICON_TYPE` to
    * accomplish a configurable button in the UI.
    */
-  @Input() set type(type: ICON_TYPE) {
+  @Input() set type(type: ICON_TYPE_STRING) {
+    // XXX this clashes with the type attribute of some html form elements, e.g. input or button
     this.setIcon(type);
   }
 
-  /**
-   * the icon provides an html fragment that is used to add SVG or text based icons.
-   */
-  icon: SafeHtml | undefined;
-
-  /**
-   * The `flip-at-rtl` class is added to the DOM for the style layer to flip the icon in RTL direction.
-   */
-  @HostBinding('class.flip-at-rtl') flipAtRtl: boolean;
-
-  /**
-   * The `flip-at-ltr` class is added to the DOM for the style layer to flip the icon in LTR direction.
-   */
-  @HostBinding('class.flip-at-ltr') flipAtLtr: boolean;
```

Comment:
Replaced these with a single `@HostBinding` for `class`. Seems to make it easier.

Reply:
We can't remove public class attributes in a minor release since it introduces a breaking change.